### ISSUE TITLE
Request Access: button does not launch popup if logged in and request from file landing page.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#4897]
Reported by user in production, see RT 264937 and reproduced here.

This works from the dataset files tab.

This is separate from the broken log in link in popup if not logged in.

Browser reports:
Uncaught TypeError: Cannot read property 'show' of undefined
    at HTMLAnchorElement.onclick (file.xhtml?persistentId=doi:10.5072/FK2/P7RCOS/KCU6BZ&version=1.0:133)

**Comment by pdurbin** [comment]
I suspect this is a duplicate of #4554.

**Comment by thaorell** [comment]
@kcondon Hi Kevin, sorry for the delay. how did you reproduce this problem? I tried create datasets and access it unauthorized but couldn't reproduce the error

**Comment by thaorell** [comment]
While I tried to reproduce the error, I found this error when trying to publish a dataset. Does it look like a separate issue? 
<img width="1323" alt="screen shot 2018-09-27 at 1 44 27 pm" src="https://user-images.githubusercontent.com/25762021/46164447-fef3c080-c25b-11e8-80f3-6f39397ba153.png">

**Comment by kcondon** [comment]
@thaorell 

It has been a while but general steps to reproduce:
1. restrict a file, enable request access, publish dataset.
2. as a separate logged in user, go to dataset with restricted file, click on file name so you end up on file landing page, click request access. Normally you would see a popup with terms and agreement/continue button but does not appear here.
3. Go back to dataset, click request access next to restricted file from there and see popup.

**Comment by thaorell** [comment]
@kcondon I got that error when I tried publishing the dataset, does it look familiar to you?

**Comment by kcondon** [comment]
@thaorell We've checked in a new version of the installer last night which at the last moment changed the default persistent identifier provider from EZID to DataCite. DataCite requires you to contact them to get an account in order to publish a dataset. The error just indicates you do not have /have not specified an account and so cannot register a PID/publish. As a workaround you could switch back to EZID manually for now using apitest as username/password.

**Comment by pdurbin** [comment]
@thaorell the support address for DataCite is support@datacite.org which you can find with more context at http://guides.dataverse.org/en/4.9.4/installation/config.html#persistent-identifiers-and-publishing-datasets

**Comment by pdurbin** [comment]
@thaorell I don't know if you're still blocked or not but heads up that over in #5297 we are trying to sort out the arrangement with DataCite for test accounts. As @kcondon indicated above, a workaround is to switch back to EZID. EZID is going away for users outside of California but I'm not sure when.

**Comment by thaorell** [comment]
@pdurbin I have been trying to contact datacite for several times to request a testing account without much success, will test with EZID

**Comment by pdurbin** [comment]
@thaorell hi! Please try the new "FAKE" DOI provider we added in Dataverse 4.10 in #5327

The docs are here:

http://guides.dataverse.org/en/4.10/developers/dev-environment.html#configure-your-development-environment-for-publishing

curl command:

    curl http://localhost:8080/api/admin/settings/:DoiProvider -X PUT -d FAKE

Does that help you get unblocked?

**Comment by adam3smith** [comment]
Was about to create a new issue, instead just here to say that this is still broken and provide some test cases:

https://demo.dataverse.org/file.xhtml?persistentId=doi:10.5072/FK2/MLFZP2/H4S7RY&version=1.0#
https://dataverse.harvard.edu/file.xhtml?persistentId=doi:10.7910/DVN/UTJ7QL/DHXMZF#

or for us on
https://data.qdr.syr.edu/file.xhtml?persistentId=doi:10.5064/F6Z60KZB/T3WH9Q&version=1.0#

**Comment by pdurbin** [comment]
@thaorell hi! Should you continue to be assigned to this issue? Please let us know. Thanks!

**Comment by mheppler** [comment]
There was an error in the browser console saying that the request access popup was missing `TypeError: PF(requestAccessPopup) is undefined` when you click on the Request Access btn on the file pg. When comparing the popup code on the dataset pg to the that on the file pg, you could see that the request popup block was missing, so I added it.

```
                    <p:dialog id="requestAccessPopup" styleClass="largePopUp" header="#{bundle['file.requestAccess']}" widgetVar="requestAccessPopup" modal="true">
                        <o:importFunctions type="edu.harvard.iq.dataverse.util.MarkupChecker" />
                        <ui:include src="file-request-access-popup-fragment.xhtml">
                            <ui:param name="workingVersion" value="#{FilePage.fileMetadata.datasetVersion}"/>
                            <ui:param name="fileDownloadService" value="#{FilePage.fileDownloadService}"/>
                        </ui:include>
                    </p:dialog>
```

Note: had to remove the `fileDownloadHelper` param from the code copied over from the dataset pg, but everything still functioned properly when testing locally.

PR forthcoming. Let me know if I am missing something bigger.
